src/commands/account/export_account/mod.rs

```diff
@@ -0,0 +1,236 @@
+/// How would you like to export the account?
+pub enum ExportAccountActions {
+    #[strum_discriminants(strum(
+        message = "using-web-wallet          - Export existing account using NEAR Wallet"
+    ))]
+    /// Export existing account using NEAR Wallet
+    UsingWebWallet(self::using_web_wallet::ExportAccountFromWebWallet),
+    #[strum_discriminants(strum(
+        message = "using-seed-phrase         - Export existing account using a seed phrase"
+    ))]
+    /// Export existing account using a seed phrase
+    UsingSeedPhrase(self::using_seed_phrase::ExportAccountFromSeedPhrase),
+    #[strum_discriminants(strum(
+        message = "using-private-key         - Export existing account using a private key"
+    ))]
+    /// Export existing account using a private key
+    UsingPrivateKey(self::using_private_key::ExportAccountFromPrivateKey),
+}
+
+pub fn get_account_key_pair_from_keychain(
+    network_config: &crate::config::NetworkConfig,
+    account_id: &near_primitives::types::AccountId,
+) -> color_eyre::eyre::Result<crate::transaction_signature_options::AccountKeyPair> {
+    let password = get_password_from_keychain(network_config, account_id)?;
+    let account_key_pair = serde_json::from_str(&password);
+    if let Ok(key_pair) = account_key_pair {
+        return Ok(key_pair);
+    }
+    Err(color_eyre::eyre::Report::msg("Error reading data"))
+}
+
+pub fn get_password_from_keychain(
+    network_config: &crate::config::NetworkConfig,
+    account_id: &near_primitives::types::AccountId,
+) -> color_eyre::eyre::Result<String> {
+    let service_name: std::borrow::Cow<'_, str> = std::borrow::Cow::Owned(format!(
+        "near-{}-{}",
+        network_config.network_name,
+        account_id.as_str()
+    ));
+    let password = {
+        let access_key_list = network_config
+            .json_rpc_client()
+            .blocking_call_view_access_key_list(
+                account_id,
+                near_primitives::types::Finality::Final.into(),
+            )
+            .wrap_err_with(|| format!("Failed to fetch access key list for {}", account_id))?
+            .access_key_list_view()?;
+
+        let res = access_key_list
+            .keys
+            .into_iter()
+            .filter(|key| {
+                matches!(
+                    key.access_key.permission,
+                    near_primitives::views::AccessKeyPermissionView::FullAccess
+                )
+            })
+            .map(|key| key.public_key)
+            .find_map(|public_key| {
+                let keyring =
+                    keyring::Entry::new(&service_name, &format!("{}:{}", account_id, public_key))
+                        .ok()?;
+                keyring.get_password().ok()
+            });
+
+        match res {
+            Some(password) => password,
+            None => {
+                return Err(color_eyre::Report::msg("No access keys found in keychain"));
+            }
+        }
```

Comment:
`wrap_err` should do the trick:

```suggestion
        access_key_list
            .keys
            .into_iter()
            .filter(|key| {
                matches!(
                    key.access_key.permission,
                    near_primitives::views::AccessKeyPermissionView::FullAccess
                )
            })
            .map(|key| key.public_key)
            .find_map(|public_key| {
                let keyring =
                    keyring::Entry::new(&service_name, &format!("{}:{}", account_id, public_key))
                        .ok()?;
                keyring.get_password().ok()
            })
            .wrap_err("No access keys found in keychain")?
```